edge-modules/MetricsCollector/MetricWithCustomTags.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Diagnostics
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public sealed class MetricWithCustomTags : Metric, IEquatable<Metric>
```

Comment:
if we are 1 more property in this class, we may need to rename this class.  You need to rename this class, e.g. "ExtendedMetric".